features/space/impl/src/main/kotlin/io/element/android/features/space/impl/leave/LeaveSpaceState.kt

```diff
@@ -18,18 +17,34 @@ data class LeaveSpaceState(
     val leaveSpaceAction: AsyncAction<Unit>,
     val eventSink: (LeaveSpaceEvents) -> Unit,
 ) {
-    val showQuickAction = selectableSpaceRooms.dataOrNull().orEmpty().any { !it.isLastAdmin }
-    val hasOnlyLastAdminRoom = selectableSpaceRooms.dataOrNull()
-        ?.let { rooms ->
-            rooms.isNotEmpty() && rooms.all { it.isLastAdmin }
-        }
-        .orFalse()
-    val numberOfSelectRooms = selectableSpaceRooms.dataOrNull().orEmpty().count { it.isSelected }
+    private val rooms = selectableSpaceRooms.dataOrNull().orEmpty()
 
-    val areAllSelected = selectableSpaceRooms.dataOrNull()
-        ?.filter { !it.isLastAdmin }
-        ?.let { rooms ->
+    /**
+     * True if we should show the quick action to select/deselect all rooms.
+     */
+    val showQuickAction = rooms
+        .any { !it.isLastAdmin }
+
+    /**
+     * True if there are rooms and they are all selected.
+     */
+    val areAllSelected = rooms
+        .filter { !it.isLastAdmin }
```

Comment:
Not a big deal, but maybe we could reuse the `rooms.filter` here for the `.any` above and the `.all` below instead of filtering the list 3 times? Since it'll be a small list (probably) maybe it's not worth it.